src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs

```diff
@@ -1,33 +1,218 @@
+            {
+                _logger.LogInformation("We could not locate a baseline for project {0}, now trying fallback Version", _options.ProjectName);
+
+                return await GetFallbackBaseline();
+            }
+
+            _logger.LogInformation("Found report of project {0} using version {1} ", _options.ProjectName, branchName);
+
+            return report;
+        }
+
+        public async Task<JsonReport> GetFallbackBaseline()
+        {
+            var report = await _dashboardClient.PullReport(_options.FallbackVersion);
+
+            if (report == null)
+            {
+                _logger.LogInformation("We could not locate a baseline for project using fallback version. Now running a complete test to establish a baseline.");
+                return null;
+            }
+            else
+            {
+                _logger.LogInformation("Found report of project {0} using version {1}", _options.ProjectName, _options.FallbackVersion);
+
+                return report;
+            }
+        }
+
+        private void UpdateMutantsWithBaseline(IEnumerable<Mutant> mutants, FileLeaf file)
+        {
+            foreach (var baselineFile in _baseline.Files)
+            {
+                var filePath = FilePathUtils.NormalizePathSeparators(baselineFile.Key);
+
+                if (filePath == file.RelativePath)
+                {
+                    foreach (var baselineMutant in baselineFile.Value.Mutants)
+                    {
+                        var baselineMutantSourceCode = GetMutantSourceCode(baselineFile.Value.Source, baselineMutant);
+
+                        IEnumerable<Mutant> matchingMutants = GetMatchingMutants(mutants, baselineMutant, baselineMutantSourceCode);
+
+                        if (matchingMutants.Count() == 1)
+                        {
+                            UpdateMutantStatusWithBaseline(baselineMutant, matchingMutants);
+                        }
+                        else
+                        {
+                            UpdateMutantStatusForNoCertainResult(matchingMutants);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void UpdateMutantStatusWithBaseline(JsonMutant baselineMutant, IEnumerable<Mutant> matchingMutants)
+        {
+            matchingMutants.First().ResultStatus = (MutantStatus)Enum.Parse(typeof(MutantStatus), baselineMutant.Status);
+            matchingMutants.First().ResultStatusReason = "Result based on previous run.";
+        }
+
+        private IEnumerable<Mutant> GetMatchingMutants(IEnumerable<Mutant> mutants, JsonMutant baselineMutant, string baselineMutantSourceCode)
+        {
+            return mutants.Where(
+            x => x.Mutation.OriginalNode.ToString() == baselineMutantSourceCode
+            && x.Mutation.DisplayName == baselineMutant.MutatorName);
+        }
+
+        private void UpdateMutantStatusForNoCertainResult(IEnumerable<Mutant> matchingMutants)
```

Comment:
It's called NoCertainResult here but StatusUnclear in ResetStatusForStatusUnclear. Pick one naming convention for this type of mutant status/result. I prefer StatusUnclear.